### PR TITLE
feat(nimbus): select required/excluded branches in audience page

### LINKED ISSUE #10029: Add branch options to required/excluded drop downs in Nimbus UI
Let’s modify the drop downs for required/excluded experiments to include options for both all branches as well as individual branches so that each experiment appears as something like:

* Experiment Title (All Branches)
* Experiment Title (Control)
* Experiment Title (Treatment)

and then includes both experiment id and branch slug when calling the V5 API.

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/EXP-4193)

### PR DESCRIPTION
**Pull request by jaredlockhart**
Because

* We can now require/exclude specific branches of experiments/rollouts

This commit

* Adds entries in the require/exclude drop downs for specific branches of each experiment/rollout

fixes #10029

image

### CODE REVIEW

**Comment by jaredlockhart** [comment]
Let's block this on https://github.com/mozilla/experimenter/issues/10026

File: experimenter/experimenter/nimbus-ui/src/components/Summary/TableAudience/index.tsx
Comment (eliserichards) [inline experimenter/experimenter/nimbus-ui/src/components/Summary/TableAudience/index.tsx:176]: Nit:
```suggestion
                    (rb) => rb.requiredExperiment,
```

File: experimenter/experimenter/nimbus-ui/src/components/PageEditAudience/FormAudience/index.tsx
Comment (eliserichards) [inline experimenter/experimenter/nimbus-ui/src/components/PageEditAudience/FormAudience/index.tsx:202]: 💯

File: experimenter/experimenter/nimbus-ui/src/components/PageEditAudience/FormAudience/index.tsx
Comment (eliserichards) [inline experimenter/experimenter/nimbus-ui/src/components/PageEditAudience/FormAudience/index.tsx:95]: Do we need all three of `id`, `slug`, and `branchSlug` here? I don't think we really need to show the `slug` to the user 🤔  wdyt?

File: experimenter/experimenter/nimbus-ui/src/components/PageEditAudience/FormAudience/index.test.tsx
Comment (eliserichards) [inline experimenter/experimenter/nimbus-ui/src/components/PageEditAudience/FormAudience/index.test.tsx:60]: Beautiful tests 🥇

File: experimenter/experimenter/nimbus-ui/src/components/Summary/TableAudience/index.tsx
Comment (jaredlockhart) [reply experimenter/experimenter/nimbus-ui/src/components/Summary/TableAudience/index.tsx:176]: I wouldn't focus on nits of this form, these variable names are so inconsequential and typescript will keep them intact.

File: experimenter/experimenter/nimbus-ui/src/components/PageEditAudience/FormAudience/index.tsx
Comment (jaredlockhart) [reply experimenter/experimenter/nimbus-ui/src/components/PageEditAudience/FormAudience/index.tsx:95]: The id is used for submitting to the API, the slug gets shown in the drop down.

**Review by brennie — APPROVED** [review]
Some suggestions

File: experimenter/experimenter/nimbus-ui/src/components/PageEditAudience/FormAudience/index.tsx
Comment (eliserichards) [reply experimenter/experimenter/nimbus-ui/src/components/PageEditAudience/FormAudience/index.tsx:95]: But why do we need to show the slug in the dropdown? Is there a reason why a user would need to see that? It seems superfluous

Comment (jaredlockhart) [reply experimenter/experimenter/nimbus-ui/src/components/PageEditAudience/FormAudience/index.tsx:95]: Oh becuase often people identify things by the slug rather than the name, since the names can be similar or ambiguous but the slugs are exact.

File: experimenter/experimenter/nimbus-ui/src/components/Summary/TableAudience/index.tsx
Comment (eliserichards) [reply experimenter/experimenter/nimbus-ui/src/components/Summary/TableAudience/index.tsx]: Ok

File: experimenter/experimenter/nimbus-ui/src/components/PageEditAudience/FormAudience/index.tsx
Comment (eliserichards) [reply experimenter/experimenter/nimbus-ui/src/components/PageEditAudience/FormAudience/index.tsx:95]: Awesome, makes sense 👍

File: experimenter/experimenter/nimbus-ui/src/components/PageEditAudience/FormAudience/index.tsx
Comment (jaredlockhart) [reply experimenter/experimenter/nimbus-ui/src/components/PageEditAudience/FormAudience/index.tsx:159]: Aha! I knew there had to be a better pattern for that thank you 💯

Comment (jaredlockhart) [reply experimenter/experimenter/nimbus-ui/src/components/PageEditAudience/FormAudience/index.tsx]: Oh these were all good suggestions but I should've reimplemented them by hand instead of merging the suggestions 😅

**Comment by jaredlockhart** [comment]
We can merge this after merging https://github.com/mozilla/experimenter/pull/10098

**Pull request merged**
